src/SAWScript/Interpreter.hs

```diff
@@ -2617,6 +2617,14 @@ primitives = Map.fromList
     Current
     [ "Legacy alternative name for `llvm_return`." ]
 
+  , prim "llvm_cast_pointer" "SetupValue -> LLVMType -> SetupValue"
```

Comment:
Can you also add mention of this new function to the [changelog](https://github.com/GaloisInc/saw-script/blob/b54197e23626bba2dc1b007eca49763d8f37c350/CHANGES.md) and [SAW manual](https://github.com/GaloisInc/saw-script/blob/b54197e23626bba2dc1b007eca49763d8f37c350/doc/manual/manual.md?plain=1)? One of the key use cases for `llvm_cast_pointer` is union types, so I think it would be worth mentioning this in the manual.

Another thing I'm wondering about: is `llvm_points_to_at_type p t v` always equivalent to `llvm_points_to (llvm_cast_pointer p t) v`? If so, I wonder if we could eventually deprecate `llvm_points_to_at_type` in favor of `llvm_points_to` + `llvm_cast_pointer`.

Reply:
Regarding `llvm_points_to_at_type`: I think you are correct that it is redundant in the presence of `llvm_cast_pointer`.  Not sure if it is worth eventually removing.